org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -88,6 +90,22 @@
 	private PreferenceManager preferenceManager;
 	protected JavaLanguageClient client;
 
+	private class ProjectPreferenceChangeListener implements IPreferenceChangeListener {
```

Comment:
I haven't seen updatePreviewFeatures being triggered through this listener. Is there a particular case for this or just an additional safety measure to ensure the change takes place ?

Reply:
just in case if the standard server starts before finishing the syntax server.